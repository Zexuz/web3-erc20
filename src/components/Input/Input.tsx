interface InputProps {
  placeholder: string;
  label: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

export const Input = ({ onChange, label, placeholder }: InputProps) => {
  return (
    <div className={"flex flex-col space-y-2"}>
      <label className={"font-poppins text-sm font-medium"}>{label}</label>
      <input
        className={"text-blue-600"}
        type="text"
        placeholder={placeholder}
        onChange={onChange}
      />
    </div>
  );
};
