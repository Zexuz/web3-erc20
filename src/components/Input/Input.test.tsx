import renderer from "react-test-renderer";
import { Input } from "./Input.tsx";

describe("Input", () => {
  it("should render", () => {
    const tree = renderer
      .create(<Input placeholder="test" label="test" onChange={() => {}} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
