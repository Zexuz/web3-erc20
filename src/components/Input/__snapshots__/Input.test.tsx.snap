// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render 1`] = `
<div
  className="flex w-full flex-col space-y-2"
>
  <label
    className="font-poppins text-sm font-medium "
  >
    test
  </label>
  <input
    className="rounded-lg border bg-blue-300 p-2.5 text-sm text-gray-200 hover:ring-2 hover:ring-blue-200 focus:border-primary focus:outline-none focus:ring-2 focus:ring-blue-400  "
    onChange={[Function]}
    placeholder="test"
    type="text"
  />
</div>
`;
